src/preamble.js

```diff
@@ -2212,7 +2224,9 @@ function integrateWasmJS(Module) {
 #if BINARYEN_ASYNC_COMPILATION
     Module['printErr']('asynchronously preparing wasm');
     addRunDependency('wasm-instantiate'); // we can't run yet
-    WebAssembly.instantiate(getBinary(), info).then(function(output) {
+    getBinaryPromise().then(function(binary) {
```

Comment:
The main .html file is supposed to download the .wasm file. At this point' is it not possible that the .wasm file download is still in progress, but has not finished yet, and therefore `getBinaryPromise()` will kick off a second parallel download redundantly?